core/pkg/service/flag-evaluation/json_codec.go

```diff
@@ -0,0 +1,95 @@
+//nolint:wrapcheck
+package service
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/bufbuild/connect-go"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/runtime/protoiface"
+)
+
+// WithJSON customizes a connect-go Client or Handler's JSON by exposing MarshalOptions, and UnmarshalOptions
+// See: https://github.com/bufbuild/connect-go/blob/main/codec.go
+// Heavily inspired by https://github.com/akshayjshah/connectproto
+func WithJSON(marshal protojson.MarshalOptions, unmarshal protojson.UnmarshalOptions) connect.Option {
+	return connect.WithOptions(
+		// mark the codec with the correct content-type
+		connect.WithCodec(&jsonCodec{name: "json", marshal: marshal, unmarshal: unmarshal}),
+		connect.WithCodec(&jsonCodec{name: "json; charset=utf-8", marshal: marshal, unmarshal: unmarshal}),
+	)
+}
+
+type jsonCodec struct {
+	name      string
+	marshal   protojson.MarshalOptions
+	unmarshal protojson.UnmarshalOptions
+}
+
```

Comment:
```suggestion
var _ connect.Codec = (*jsonCodec)(nil)
```

This is a compile time check to see if `jsonCodec` implements `connect.Codec` interface

Reply:
Connect internals also do the same - https://github.com/bufbuild/connect-go/blob/main/codec.go#L146C1-L146C37